### ISSUE TITLE
Allow users to run framework without docker

### ISSUE THREAD

**Issue by philwinder** [mesos/elasticsearch#334]
Host the executor JAR (the scheduler JAR has the executor JAR bundled within) on the scheduler, pass URI to TaskInfo and allow Mesos to download and start the executor.

See code from a long time ago which implemented this, which was then deleted.

Update this issue when tasks are established.

**Comment by philwinder** [comment]
Wooh working. Code was resurrected from here:
https://github.com/mesos/elasticsearch/blob/d492379352957b749c606b032e410ac89b055288/scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/ElasticsearchScheduler.java

New code is in the enhancement/334-UseJars branch.

Jars were built and scp'ed onto master DCOS machine. SSH'ed into master, then scp'ed to every DCOS slave.

I used the following JSON:
```
{
  "id": "elasticsearch-jar",
  "cpus": 0.5,
  "mem": 512,
  "instances": 1,
  "cmd": "/opt/mesosphere/bin/java -jar /home/core/elasticsearch-mesos-scheduler-0.4.3.jar --javaHome /opt/mesosphere/bin/java --frameworkName esjar --elasticsearchClusterName esjarcluster --frameworkUseDocker false --zookeeperMesosUrl zk://52.18.166.109:2181",
  "env": {
    "JAVA_OPTS": "-Xms128m -Xmx256m"
  },
  "ports": [31100],
  "requirePorts": true
}
```

Resulted in working cluster: http://52.18.166.109/service/esjar/#/

Remaining tasks:
[ ] - Try and remove that nasty java home with something more generic. Maybe a new configuration option to allow users to pass their own env vars, then statically hard code $JAVA_HOME into the commandInfo.
[ ] - Write system test.

**Comment by frankscholten** [comment]
Looks good. In the system tests the scheduler should also not run in Docker. Let's create a builder for the Scheduler on which we can configure it to use Docker or not.

**Comment by frankscholten** [comment]
When running this branch on 0.2.12 the tasks won't start because MiniMesos has an older JVM version.

Therefore we should host the JVM on the scheduler webserver so it will be downloaded and unpacked in the sandbox when tasks are launched.

**Comment by frankscholten** [comment]
We can then remove the Java home flag because it will always run Java from the current directory, which is the sandbox.

**Comment by frankscholten** [comment]
This is not probably not going to work because we need the entire JRE distribution not just the Java binary.

**Comment by philwinder** [comment]
Fixed in #355
